kafka-connect/src/main/java/com/hazelcast/jet/contrib/connect/KafkaConnectSources.java

```diff
@@ -48,43 +51,91 @@ private KafkaConnectSources() {
+     * contains the JARs as a job resource via {@link com.hazelcast.jet.config.JobConfig#addJar(URL)}
+     * or {@link com.hazelcast.jet.config.JobConfig#addJarsInZip(URL)}
+     * respectively.
      * <p>
-     * After that you can use the Kafka Connect connector with the configuration
-     * parameters as you'd using it with Kafka. Hazelcast Jet will drive the
-     * Kafka Connect connector from the pipeline and the records will be available
-     * to your pipeline as {@link SourceRecord}s.
+     * After that you can use the Kafka Connect connector with the
+     * configuration parameters as you'd using it with Kafka. Hazelcast
+     * Jet will drive the Kafka Connect connector from the pipeline and
+     * the records will be available to your pipeline as {@link SourceRecord}s.
      * <p>
-     * In case of a failure; this source keeps track of the source partition
-     * offsets, it will restore the partition offsets and resume the consumption
-     * from where it left off.
+     * In case of a failure; this source keeps track of the source
+     * partition offsets, it will restore the partition offsets and
+     * resume the consumption from where it left off.
      * <p>
-     * Hazelcast Jet will instantiate a single task for the specified source in
-     * the cluster.
+     * Hazelcast Jet will instantiate a single task for the specified
+     * source in the cluster.
      *
      * @param properties Kafka connect properties
      * @return a source to use in {@link com.hazelcast.jet.pipeline.Pipeline#readFrom(StreamSource)}
      */
     public static StreamSource<SourceRecord> connect(Properties properties) {
+        return connect(properties, () -> null,
+                (IGNORED, record) -> record.timestamp() == null ? 0 : record.timestamp(),
+                (BiFunctionEx<Void, SourceRecord, SourceRecord>) (IGNORED, record) -> record
+        );
+    }
+
+    /**
+     * A generic Kafka Connect source provides ability to plug any Kafka
+     * Connect source for data ingestion to Jet pipelines.
+     * <p>
+     * You need to add the Kafka Connect connector JARs or a ZIP file
+     * contains the JARs as a job resource via {@link com.hazelcast.jet.config.JobConfig#addJar(URL)}
+     * or {@link com.hazelcast.jet.config.JobConfig#addJarsInZip(URL)}
+     * respectively.
+     * <p>
+     * After that you can use the Kafka Connect connector with the
+     * configuration parameters as you'd using it with Kafka. Hazelcast
+     * Jet will drive the Kafka Connect connector from the pipeline and
+     * the records will be available to your pipeline as {@link SourceRecord}s.
+     * <p>
+     * In case of a failure; this source keeps track of the source
+     * partition offsets, it will restore the partition offsets and
+     * resume the consumption from where it left off.
+     * <p>
+     * Hazelcast Jet will instantiate a single task for the specified
+     * source in the cluster.
```

Comment:
can we have a few words in the javadoc explaining the usage of added projection functions. I was not particularly sure why we need a state object for example.

Reply:
Yes. Definitely. Strictly speaking the practical role of the projection state is to have a separate instance of Jackson `ObjectMapper` in each source processor instance.